### PR TITLE
Add score statistic tracking (osu!).

### PR DESCRIPTION
**Pull request by peppy**
- [x] Depends on #648

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Scoring/OsuScoreProcessor.cs [hunk 1]
```diff
@@ -26,12 +28,46 @@ protected override void Reset()
...
         {
             if (judgement != null)
             {
+                if (judgement.Result != HitResult.None)
+                {
+                    int count;
+                    if (scoreResultCounts.TryGetValue(judgement.Score, out count))
+                        scoreResultCounts[judgement.Score] = count + 1;
+                    else
+                        scoreResultCounts[judgement.Score] = 0;
```
Comment (smoogipooo) [inline osu.Game.Rulesets.Osu/Scoring/OsuScoreProcessor.cs]: Shouldn't this and the same line for comboResultCounts be `= 1`?

Comment (peppy) [inline osu.Game.Rulesets.Osu/Scoring/OsuScoreProcessor.cs]: yes

File: osu.Game/Rulesets/Scoring/Score.cs [hunk 2]
```diff
@@ -49,6 +48,6 @@ public class Score
         [JsonProperty(@"created_at")]
         public DateTime Date;
 
-        public virtual IEnumerable<ScoreStatistic> Statistics => new ScoreStatistic[] { };
+        public Dictionary<string, dynamic> Statistics = new Dictionary<string, dynamic>();
```
Comment (smoogipooo) [inline osu.Game/Rulesets/Scoring/Score.cs]: Seems like you removed ScoreStatistic but it isn't removed from the project.

File: osu.Game/Screens/Play/Player.cs [hunk 3]
```diff
@@ -82,15 +84,17 @@ private void load(AudioManager audio, BeatmapDatabase beatmaps, OsuConfigManager
...
                 {
                     // Default to the beatmap ruleset
-                    ruleset = Beatmap.BeatmapInfo.Ruleset.CreateInstance();
-                    HitRenderer = ruleset.CreateHitRendererWith(Beatmap);
+                    ruleset = Beatmap.BeatmapInfo.Ruleset;
                 }
+
+                rulesetInstance = ruleset.CreateInstance();
+
+                HitRenderer = rulesetInstance.CreateHitRendererWith(Beatmap);
```
Comment (smoogipooo) [inline osu.Game/Screens/Play/Player.cs]: Nope, I did it this way because HitRenderer internally throws if it can't convert the beatmap.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**